similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -177,6 +194,39 @@ public int compare(PatientSimilarityView o1, PatientSimilarityView o2)
         return results;
     }
 
+    private boolean filterPatient(Patient matchPatient, Family family, EntityReference refOwner,
+        String requiredConsentId)
+    {
+        if (matchPatient == null) {
+            // Leftover patient in the index, should be removed
+            return true;
+        }
+        // filter out patients from the same family
+        if (family != null && family.isMember(matchPatient)) {
+            return true;
+        }
+        // filter out patients of the same owner
+        if (refOwner.equals(this.accessHelper.getOwner(matchPatient).getUser())) {
```

Comment:
This may trigger a `NullPointerException` if the owner is `null`, which shouldn't happen but as we've seen yesterday, it may if there are other bugs.